src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -874,6 +864,241 @@ TEST_F(TestCASESession, Sigma1ParsingTest)
+        CASESession::ParsedSigma1 parseParams;
+
+        // Round Trip Test: Sigma1 with Session Resumption
+        // Encode Sigma1 with Resumption, parse it and and verify with original values
+        chip::SessionResumptionStorage::ResumptionIdStorage resumptionId;
+
+        EXPECT_EQ(chip::Crypto::DRBG_get_bytes(resumptionId.data(), resumptionId.size()), CHIP_NO_ERROR);
+        EXPECT_EQ(chip::Crypto::DRBG_get_bytes(&encodeParams.initiatorResume1MIC[0], sizeof(encodeParams.initiatorResume1MIC)),
+                  CHIP_NO_ERROR);
+
+        encodeParams.resumptionId               = ByteSpan(resumptionId.data(), resumptionId.size());
+        encodeParams.initiatorResumeMICSpan     = ByteSpan(encodeParams.initiatorResume1MIC);
+        encodeParams.sessionResumptionRequested = true;
+
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma1(msg2, encodeParams));
+
+        // Encode and Parse Round Trip Test
+        tlvReader.Init(std::move(msg2));
+
+        EXPECT_EQ(CHIP_NO_ERROR, ParseSigma1(tlvReader, parseParams));
+
+        // RoundTrip
+        EXPECT_TRUE(parseParams.initiatorRandom.data_equal(encodeParams.initiatorRandom));
+        EXPECT_EQ(parseParams.initiatorSessionId, encodeParams.initiatorSessionId);
+        EXPECT_TRUE(parseParams.destinationId.data_equal(encodeParams.destinationId));
+        EXPECT_TRUE(parseParams.initiatorEphPubKey.data_equal(
+            ByteSpan(encodeParams.pEphPubKey->ConstBytes(), encodeParams.pEphPubKey->Length())));
+
+        EXPECT_TRUE(parseParams.resumptionId.data_equal(encodeParams.resumptionId));
+        EXPECT_TRUE(parseParams.initiatorResumeMICSpan.data_equal(encodeParams.initiatorResumeMICSpan));
+        EXPECT_TRUE(parseParams.initiatorMrpParamsPresent);
+    }
+    // Release EphemeralKeyPair
+    gDeviceOperationalKeystore.ReleaseEphemeralKeypair(EphemeralKey);
+}
+
+TEST_F(TestCASESession, EncodeSigma2Test)
+{
+    CASESession::EncodeSigma2Inputs encodeParams;
+    constexpr uint8_t kEncrypted2datalen = 100U;
+
+    EXPECT_EQ(chip::Crypto::DRBG_get_bytes(&encodeParams.responderRandom[0], sizeof(encodeParams.responderRandom)), CHIP_NO_ERROR);
+    encodeParams.responderSessionId = 7315;
+
+    // Generate Ephemeral Public Key
+    Crypto::P256Keypair * EphemeralKey = gDeviceOperationalKeystore.AllocateEphemeralKeypairForCASE();
+    ASSERT_NE(EphemeralKey, nullptr);
+    EXPECT_EQ(CHIP_NO_ERROR, EphemeralKey->Initialize(ECPKeyTarget::ECDH));
+    encodeParams.pEphPubKey = &EphemeralKey->Pubkey();
+
+    // TBEData2Encrypted
+    encodeParams.encrypted2Length = kEncrypted2datalen + CHIP_CRYPTO_AEAD_MIC_LENGTH_BYTES;
+    encodeParams.msgR2Encrypted.Alloc(encodeParams.encrypted2Length);
+
+    // responder Session Parameters
+    ReliableMessageProtocolConfig MRPConfig = GetDefaultMRPConfig();
+    encodeParams.responderMrpConfig         = &MRPConfig;
+
+    {
+        System::PacketBufferHandle msg;
+        EXPECT_EQ(CHIP_NO_ERROR, EncodeSigma2(msg, encodeParams));
```

Comment:
Again, if EncodeSigma2 does not touch state it should be static, and if it does this needs to be done on a clean CASESession?